File: app/src/main/java/com/kakeibo/data/disk/SubscriptionStatusDao.java
//package com.kakeibo.data.disk;
//
//import androidx.lifecycle.LiveData;
//import androidx.room.Dao;
//import androidx.room.Insert;
//import androidx.room.OnConflictStrategy;
//import androidx.room.Query;
//
//import com.kakeibo.data.SubscriptionStatus;
//
//import java.util.List;
//
//@Dao
//interface SubscriptionStatusDao {
//    @Query("SELECT * FROM subscriptions")
//    LiveData<List<SubscriptionStatus>> getAll();
//
//    @Insert(onConflict = OnConflictStrategy.REPLACE)
//    void insertAll(List<SubscriptionStatus> comments);
//
//    @Query("DELETE FROM subscriptions")
//    void deleteAll();
//}

File: app/src/main/java/com/kakeibo/data/disk/SubscriptionStatusDao.kt
package com.kakeibo.data.disk

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.kakeibo.data.SubscriptionStatus

@Dao
interface SubscriptionStatusDao {
    @Query("SELECT * FROM subscriptions")
    fun getAll(): LiveData<List<SubscriptionStatus>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(comments: List<SubscriptionStatus>)

    @Query("DELETE FROM subscriptions")
    fun deleteAll()
}